generator/test_case_generator.go

```diff
@@ -36,65 +36,72 @@ type testConfig struct {
 	// e.g. statsd can have a multiple terraform module sets for difference test scenarios (ecs, eks or ec2)
 	testDir      string
 	terraformDir string
+	// target specific OS(es) so that only limited targets are tested. doesn't apply to containers
+	// e.g. ["rhel8"] or ["rhel8", "ubuntu-20.04"]
+	targetOs map[string]struct{}
 }
 
 // you can't have a const map in golang
 var testTypeToTestConfig = map[string][]testConfig{
 	"ec2_gpu": {
-		{"./test/nvidia_gpu", ""},
+		{"./test/nvidia_gpu", "", nil},
```

Comment:
we don't need to specify `nil` here because that's the default value of a map

Reply:
Huh. [TIL](https://go.dev/play/p/XOgSO3B3mkq). Okay, well then personally I would rather put names on the struct values, rather than use positional arguments, but this is non blocking.